src/core/meshcop/border_agent.cpp

```diff
@@ -227,10 +227,17 @@ void BorderAgent::HandleCoapResponse(void *               aContext,
     Coap::Message *      message        = NULL;
     otError              error;
 
-    VerifyOrExit((message = NewMeshCoPMessage(instance.Get<Coap::CoapSecure>())) != NULL, error = OT_ERROR_NO_BUFS);
     SuccessOrExit(error = aResult);
+    VerifyOrExit((message = NewMeshCoPMessage(instance.Get<Coap::CoapSecure>())) != NULL, error = OT_ERROR_NO_BUFS);
+
+    SuccessOrExit(error = forwardContext.ToHeader(*message, response->GetCode()));
```

Comment:
is it necessary to move this block above?

Reply:
Just want to forward whatever response from leader even if it is corrupted. But currently, it cannot achieve this because the state TLV and Session ID TLV checking will directly exit if failed. 

Reverted.